### ISSUE TITLE
After trying to save world from GUI, Gazebo freezes

### ISSUE THREAD

**Issue by gauravbhosale** [osrf/gazebo#2866]
Basically after trying to save, Gazebo freezes, and this message appears in the terminal:
> GtkDialog mapped without a transient parent. This is discouraged.
![gazeboFreezes](https://user-images.githubusercontent.com/30625216/97276840-bd300c80-185d-11eb-96f0-311591f9191b.png)

This issue seen for Gazebo 9, 10 and 11 versions.
Is there any workaround for this?

**Comment by chapulina** [comment]
I just tried this here, and I also see the message, but Gazebo is not frozen. How did you install Gazebo? Is there more information you can provide about your setup to help others reproduce it?

**Comment by gauravbhosale** [comment]
1] I have used **Gazebo-Ubuntu installation** steps mentioned in the following page
http://gazebosim.org/tutorials?tut=install_ubuntu
The Gazebo installation is performed on **Ubuntu 18.04 VM**.

2] Further, I launched Gazebo using 'gazebo --verbose' command

3] After launching Gazebo, I tried to save world using Gazebo GUI options, **File->Save world as**

4] Once I clicked on 'Save world as', Gazebo freezes. In this state, the simulation time keep on increasing but I can not click on any object/button from Gazebo GUI. 
Also, 'Save world as' should provide option of saving world or selecting folder/path, but that didn't happen after click. 
So, basically, it look like freeze state or 'Save world as' is not working as expected.

**Comment by SdKay** [comment]
I have same problem in a **remote xrdp server** and could be fixed by running gazebo with **sudo**
But in my **local computer** it works well
And don't know how to fix this in other ways..becase not everyone have sudo permissions

**Comment by Joggel1** [comment]
I have the same problem. Also as SdKay mentioned, it works when running gazebo with sudo. Interesting is, that the 'safe world as'-window appears, when I use ctrl+c (but I still can't save the world then):

https://user-images.githubusercontent.com/59475687/103637281-89532080-4f4b-11eb-88bd-798636ac2e63.mp4

**Comment by BingyiSu** [comment]
I have the same problem and have you guys solved this problem?

**Comment by Joggel1** [comment]
I accidently found out that if you minimze the gazebo window and maximize it again, the save-window appears, but after every action (e.g. click on a folder) I have to minimize and maximize again. So somehow this window just doesn't update its view.

**Comment by tim-fan** [comment]
I'm also seeing this, on Gazebo version 11.3.0 (on Ubuntu 20.04.2 LTS).

I've noted that the freeze only occurs when saving a world that contains one of my own custom models. In contrast I can save the world without issue if the world contains only models from models.gazebosim.org or fuel.ignitionrobotics.org.

This seems a little different from the issues described by @gauravbhosale and @Joggel1 , as they are not importing custom models. 

As per @Joggel1's suggestion, I can save the world successfully by repeatedly minimising then maximising before each click in the dialogue box.

**Comment by liepgp** [comment]
I have the same problem
- Fresh installed Ubuntu 20.04.2 LTS in VirtualBox
- Gazebo installed from software center

Minimizing and and maximizing makes the save window to appear
Running with sudo, it works well

**Comment by trejkev** [comment]
Just open it with sudo, e.g. 'sudo gazebo name_of_my_arena.world'. It worked pretty well for me using Ubuntu 18.

**Comment by zachpoh** [comment]
Is there any other solutions other than "sudo gazebo"? "sudo gazebo" does work but only allowing me to save the file outside of computer. I can't seem to find the file that I saved nonetheless. I am currently running ubuntu 20.04 on VirtualBox, even on dual boot the same problem arise.

**Comment by chongjeelee** [comment]
i also have the same problem,anyone solved this ?

**Comment by PhehJingJie** [comment]
> I accidently found out that if you minimze the gazebo window and maximize it again, the save-window appears, but after every action (e.g. click on a folder) I have to minimize and maximize again. So somehow this window just doesn't update its view.

This works for me, thanks so much @Joggel1 !

**Comment by co4erol** [comment]
> Just open it with sudo, e.g. 'sudo gazebo name_of_my_arena.world'. It worked pretty well for me using Ubuntu 18.

you saved my life 👍

**Comment by Beryl-Zong** [comment]
I've met another problem, when I saved the world use the method described above and try to load the world, gazebo can only show empty world without the model I saved in my_world.world.
Anyone met this?

**Comment by yabdulra** [comment]
> I've met another problem, when I saved the world use the method described above and try to load the world, gazebo can only show empty world without the model I saved in my_world.world. Anyone met this?

What models are you importing to your world? I had similar problem and found out that it happened because I was using some models I found online. Worked well with models from gazebo online library and the ones I created myself.

**Comment by ashwanthrkumar** [comment]
> I accidently found out that if you minimze the gazebo window and maximize it again, the save-window appears, but after every action (e.g. click on a folder) I have to minimize and maximize again. So somehow this window just doesn't update its view.

Thank you, issue cleared following this

**Comment by ddowling** [comment]
I have found the same problem running Ubuntu 22 with a Gnome desktop. I seems to be related to the Gnome file manager. I found a workaround as follows:
```
XDG_CURRENT_DESKTOP=kde gazebo world.sdf
```
This presents a KDE style dialog for "Save As..." that seems to avoid the problem.

**Comment by Kavinaya7** [comment]
Use Sudo gazebo, when starting gazebo
